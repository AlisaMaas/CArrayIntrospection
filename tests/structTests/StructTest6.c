/**
* Check that the results from multiple
* functions are correctly collated into 
* a single result per struct element,
* regardless of where it's found.
**/

struct foo {
	int a;
	double b;
	struct foo *c;
};

struct bar {
	struct foo first;
	int x;
	double y;
	struct bar *z;
};

void accessBar(struct bar b) {
	b.x = 5;
	b.y = 4.5;
	b.z = &b; 
}

void accessFoo(struct foo f) {
	f.a = 32;
	f.b = 1.2;
	f.c = &f;
}

void accessBoth(struct foo f, struct bar b) {
	b.x = f.a;
	b.y = f.a * f.b;
	b.z = &b;
	b.first = f;
}

void accessArrays(struct foo *f, struct bar *b) {
	for (int i = 0; i < 40; i++) {
		f[i] = f[0];
	}
	f[2] = f[0];
	b[200] = b[0];
	b[0] = b[0];
	b[1] = b[0];
	
}

void accessArrayStructElements(struct foo *f, struct bar *b) {
	for (int i = 0; i < 40; i++) {
		f[i].a = 9;
	}
	f[2].b = 2.2;
	b[200].first = f[0];
	b[0].x = 3;
	b[1].y = 3.3;
	b[4].z = b;
	
}