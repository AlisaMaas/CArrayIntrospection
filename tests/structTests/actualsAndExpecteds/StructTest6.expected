struct.bar:
	element 0 accessed at first
	element 0 accessed at first
	element 1 accessed at x
	element 1 accessed at x
	element 1 accessed at x
	element 2 accessed at y
	element 2 accessed at y
	element 2 accessed at y
	element 3 accessed at z
	element 3 accessed at z
	element 3 accessed at z
struct.foo:
	element 0 accessed at a
	element 0 accessed at a
	element 0 accessed at a1
	element 0 accessed at a
	element 1 accessed at b
	element 1 accessed at b2
	element 1 accessed at b2
	element 2 accessed at c
